### PR TITLE
feat: homepage

### PR DESCRIPTION
**Pull request by maitrakhatri**
# features added
- basic homepage
- fetch and show categories from db
- when click on category redirect to products page and only show products to that category
- added routes
- click on hero product and sidekick products will open the product listing page

# demo
https://user-images.githubusercontent.com/54719132/160362292-80c2583a-ff4d-4d90-94a1-5bdaa2afbceb.mp4

### CODE REVIEW

**Review by anupkgurung — APPROVED** [review]
For images you could also use any online host which would be faster when loading your app.
Rest seems good and clear

File: gozo-stores/src/components/Thumbnail.jsx [hunk 1]
```diff
@@ -2,12 +2,12 @@ import "./thumbnail.css"
...
     return (
-        <a href="">
-            <div class="category">
-                <img src={props.src} alt={props.name} />
-                <div class="fade"></div>
-                <div class="cat-name">{props.name}</div>
-            </div>
-        </a>
+        
+        <div class="category" onClick={props.onClick}>
```
Comment (rohanmathur91) [inline gozo-stores/src/components/Thumbnail.jsx:6]: suggestion: you can destructure the props on line number 3 itself

File: gozo-stores/src/pages/homepage.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import "./homepage.css"
+import Thumbnail from "../components/Thumbnail";
+import axios from "axios"
+import Navbar from "../components/Navbar"
+import { useEffect, useState } from "react";
+import Sidekick from "../components/SidekickCard";
+import { useFilter } from "../context/filter-context";
```
Comment (rohanmathur91) [inline gozo-stores/src/pages/homepage.jsx:7]: try barrel exports

**Review by rohanmathur91 — APPROVED** [review]
Added few suggestions
- file structure is good
- Wherever possible try to add default props also as a fallback

File: gozo-stores/src/pages/homepage.jsx (hunk 2 above)
Comment (maitrakhatri) [reply gozo-stores/src/pages/homepage.jsx:7]: heard of it for the first time, I'll read about it
edit: yes I know how to do it, just didn't know the proper name

**Pull request merged**